src/Orleans/AssemblyLoader/AssemblyProcessor.cs

```diff
@@ -145,17 +145,24 @@ private static void ProcessAssembly(Assembly assembly)
             // Process each type in the assembly.
             foreach (TypeInfo type in assemblyTypes)
             {
-                string typeName = type.FullName;
-                if (Logger.IsVerbose3)
+                try
                 {
-                    Logger.Verbose3("Processing type {0}", typeName);
+                    string typeName = type.FullName;
+                    if (Logger.IsVerbose3)
+                    {
+                        Logger.Verbose3("Processing type {0}", typeName);
+                    }
+                    if (shouldProcessSerialization)
+                    {
+                        SerializationManager.FindSerializationInfo(type);
+                    }
+    
+                    GrainFactory.FindSupportClasses(type);
                 }
-                if (shouldProcessSerialization)
+                catch (Exception exception)
                 {
-                    SerializationManager.FindSerializationInfo(type);
+                    Logger.Error(ErrorCode.SerMgr_TypeRegistrationFailure, $"Failed to load type {type.FullName} in assembly {assembly.FullName}.", exception);
```

Comment:
please don't use c# 6 features yet, as this will break some of our CI environments. We intend to migrate soon, but in the meantime this would completely stall our functional tests and load/reliability tests.